Issue with Table class

The issue is in this line 

https://github.com/bluesky/ophyd-async/blame/e4b18e58a90d6de3b309e20557ed6994fa3a19d2/src/ophyd_async/core/_table.py#L49


The type checking issue is this 
```
/ophyd-async/src/ophyd_async/core/_table.py
/ophyd-async/src/ophyd_async/core/_table.py:49:61 - error: No overloads for "array" match the provided arguments (reportCallIssue)
/ophyd-async/src/ophyd_async/core/_table.py:49:80 - error: Argument of type "dict[str, Any]" cannot be assigned to parameter "dtype" of type "DTypeLike" in function "array"
    Type "dict[str, Any]" is not assignable to type "DTypeLike"
      "dict[str, Any]" is not assignable to "dtype[Any]"
      "dict[str, Any]" is not assignable to "None"
      "dict[str, Any]" is not assignable to "type[Any]"
      "dict[str, Any]" is incompatible with protocol "_SupportsDType[dtype[Any]]"
        "dtype" is not present
      "dict[str, Any]" is not assignable to "str"
      "dict[str, Any]" is not assignable to "tuple[_DTypeLikeNested, int]"
```
Stack Trace for the test failure
```
line 146, in test_malformed_trigger_in_seq_table
    assert (
AssertionError: assert "Input should be 'Immediate', 'BITA=0', 'BITA=1', 'BITB=0', 'BITB=1', 'BITC...' [type=enum, input_value='A', input_type=str]" in "<ExceptionInfo 1 validation error for SeqTable\ntrigger.0\n  Input should be 'Immediate', 'BITA=0', 'BITA=1', 'BITB=0', 'BITB=1', 'BITC... [type=enum, input_value='A', input_type=str]\n    For further information visit https://errors.pydantic.dev/2.10/v/enum tblen=3>"
 +  where "<ExceptionInfo 1 validation error for SeqTable\ntrigger.0\n  Input should be 'Immediate', 'BITA=0', 'BITA=1', 'BITB=0', 'BITB=1', 'BITC... [type=enum, input_value='A', input_type=str]\n    For further information visit https://errors.pydantic.dev/2.10/v/enum tblen=3>" = str(<ExceptionInfo 1 validation error for SeqTable\ntrigger.0\n  Input should be 'Immediate', 'BITA=0', 'BITA=1', 'BITB=0', 'BITB=1', 'BITC... [type=enum, input_value='A', input_type=str]\n    For further information visit https://errors.pydantic.dev/2.10/v/enum tblen=3>)

```

## Acceptance Criteria
- Type checking issue and test_malformed_trigger_in_seq_table is resolved

--- a/src/ophyd_async/core/_table.py
+++ b/src/ophyd_async/core/_table.py
@@ -1,6 +1,6 @@
 from __future__ import annotations
 
-from collections.abc import Sequence
+from collections.abc import Callable, Sequence
 from typing import Annotated, Any, TypeVar, get_origin
 
 import numpy as np
@@ -17,6 +17,13 @@ def _concat(value1, value2):
         return np.concatenate((value1, value2))
     else:
         return value1 + value2
+
+
+def _make_default_factory(dtype: np.dtype) -> Callable[[], np.ndarray]:
+    def numpy_array_default_factory() -> np.ndarray:
+        return np.array([], dtype)
+
+    return numpy_array_default_factory
 
 
 class Table(BaseModel):
@@ -45,9 +52,7 @@ class Table(BaseModel):
                     NpArrayPydanticAnnotation.factory(
                         data_type=dtype.type, dimensions=1, strict_data_typing=False
                     ),
-                    Field(
-                        default_factory=lambda dtype=dtype: np.array([], dtype=dtype)
-                    ),
+                    Field(default_factory=_make_default_factory(dtype)),
                 ]
             elif get_origin(anno) is Sequence:
                 new_anno = Annotated[anno, Field(default_factory=list)]

--- a/tests/fastcs/panda/test_trigger.py
+++ b/tests/fastcs/panda/test_trigger.py
@@ -145,7 +145,7 @@ def test_malformed_trigger_in_seq_table():
         full_seq_table(["A"])
     assert (
         "Input should be 'Immediate', 'BITA=0', 'BITA=1', 'BITB=0', 'BITB=1', "
-        "'BITC...' [type=enum, input_value='A', input_type=str]"
+        "'BITC... [type=enum, input_value='A', input_type=str]"
     ) in str(exc)
 
     # Pydantic is able to infer type from these

